datastore/datastore-utils/src/test/groovy/org/orbisgis/datastore/datastoreutils/FeatureIteratorUtilsTest.groovy

```diff
@@ -0,0 +1,19 @@
+package org.orbisgis.datastore.datastoreutils
+
+import org.geotools.data.shapefile.ShapefileDataStore
+import org.junit.jupiter.api.Test
+/**
+ * Test class dedicated to {@link FeatureIteratorUtils}.
+ *
+ * @author Erwan Bocher (CNRS 2020)
+ * @author Sylvain PALOMINOS (UBS chaire GEOTERA 2020)
+ */
+class FeatureIteratorUtilsTest {
+
+    @Test
+    void eachTest() {
+        def ds = new ShapefileDataStore(this.getClass().getResource("landcover2000.shp"))
+        ds.landcover2000.features.features().each
```

Comment:
It would be nice to overwrite by 

ds.landcover2000.iterator.features().each or ds.landcover2000.featureIterator.features().each
or do 
 ds.landcover2000.featuresIterator().each

Reply:
The convention naming in geotools is bit confusing because you can have

SimpleFeatureSource.getFeatures().features();

It 'd be better to have

SimpleFeatureSource.getFeatureCollection().getFeatureIterator();

Reply:
#### 1rst comment
The geotools structure is `DataSource.FeatureSource.FeatureCollection.FeatureIterator`.
So in this example : 
 - `ds.landcover2000.iterator.features().each{}`, the `iterator` is a `FeatureCollection` 
 - `ds.landcover2000.featureIterator.features().each` the `featureIterator` is a `FeatureCollection`

What do you think about : 
``` groovy
ds.landcover2000.iterator.features().each {...}
//replaced by
ds.landcover2000.collection.features().each {...}

ds.landcover2000.featureIterator.features().each {...}
//replaced by
ds.landcover2000.featureCollection.features().each {...}

ds.landcover2000.featuresIterator().each {...}
//replaced by
ds.landcover2000.features.each {...}
```

#### 2nd comment
Ok for the syntax `FeatureSource.getFeatureCollection().getFeatureIterator();` but the geootools syntax will still be available.

Reply:
2nd comment APPROVED :+1: